packages/compiler-dom/__tests__/index.spec.ts

```diff
@@ -10,4 +10,20 @@ describe('compile', () => {
 
     expect(code).toMatchSnapshot()
   })
+
+  // #3001
```

Comment:
When I run the tests it doesn't use the decoder in `decodeHtmlBrowser.ts`, it uses the decoder in `decodeHtml` instead. I'm not sure whether there's an easy way to trigger the `__BROWSER__` flag but at the moment it doesn't seem this test is really testing the code that has changed.

I experimented with writing a new spec in `packages/compiler-dom/__tests__/decodeHtmlBrowser.spec.ts` with the following contents. It's a bit rough but I think it tests the major points:

```js
import { decodeHtmlBrowser } from '../src/decodeHtmlBrowser'

describe('decodeHtmlBrowser', () => {
  it('should decode HTML correctly', () => {
    expect(decodeHtmlBrowser(' abc  123 ')).toBe(' abc  123 ')

    expect(decodeHtmlBrowser('&')).toBe('&')
    expect(decodeHtmlBrowser('&amp;')).toBe('&')
    expect(decodeHtmlBrowser('&amp;amp;')).toBe('&amp;')

    expect(decodeHtmlBrowser('<')).toBe('<')
    expect(decodeHtmlBrowser('&lt;')).toBe('<')
    expect(decodeHtmlBrowser('&amp;lt;')).toBe('&lt;')

    expect(decodeHtmlBrowser('>')).toBe('>')
    expect(decodeHtmlBrowser('&gt;')).toBe('>')
    expect(decodeHtmlBrowser('&amp;gt;')).toBe('&gt;')

    expect(decodeHtmlBrowser('&nbsp;')).toBe('\u00a0')
    expect(decodeHtmlBrowser('&quot;')).toBe('"')
    expect(decodeHtmlBrowser('&apos;')).toBe('\'')

    expect(decodeHtmlBrowser('&Eacute;')).toBe('\u00c9')
    expect(decodeHtmlBrowser('&#xc9;')).toBe('\u00c9')
    expect(decodeHtmlBrowser('&#201;')).toBe('\u00c9')

    // #3001
    expect(decodeHtmlBrowser('<strong>Text</strong>')).toBe('<strong>Text</strong>')
    expect(decodeHtmlBrowser('<strong>&amp;</strong>')).toBe('<strong>&</strong>')
    expect(decodeHtmlBrowser('<strong>&lt;strong&gt;&amp;&lt;/strong&gt;</strong>')).toBe('<strong><strong>&</strong></strong>')
  })
})
```

I tried running the same spec against `decodeHtml`, to confirm that the proposed changes will be consistent between the two decoders. All the assertions I've made here also pass with `decodeHtml`.

I also confirmed that it fails against the current `master` version of `decodeHtmlBrowser.ts`.

Reply:
It's a very good idea, I have updated it.
Thanks for your suggestions.